src/Azure/Queues/AzureServiceBusQueue.cs

```diff
@@ -133,40 +149,50 @@ public class AzureServiceBusQueue<T> : QueueBase<T> where T : class {
                     if (!queueEntry.IsAbandoned && !queueEntry.IsCompleted)
                         await queueEntry.AbandonAsync().AnyContext();
                 }
+            }, new OnMessageOptions {
+                AutoComplete = false
             });
```

Comment:
This will tell Service Bus not to autocomplete... Autocompletion is controlled by `queueEntry.CompleteAsync()` and `queueEntry.AbandonAsync()` instead in order to trigger the events.